bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/config/CredentialOptions.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.config;
+
+import com.google.common.base.Preconditions;
+
+public class CredentialOptions {
+  public static final String SERVICE_ACCOUNT_JSON_ENV_VARIABLE = "GOOGLE_APPLICATION_CREDENTIALS";
+  protected static final Logger LOG = new Logger(CredentialOptions.class);
+
+  public enum CredentialType {
+    Json, MetadataServiceAccount, None, P12,
+  }
+
+  /**
+   * Credential options to get the credentials from a Json file set in the
+   * GOOGLE_APPLICATION_CREDENTIALS environment variable -
+   * https://developers.google.com/identity/protocols/application-default-credentials
+   */
+  public static CredentialOptions jsonCredential() {
+    return new CredentialOptions(CredentialType.Json);
+  }
+
+  public static boolean hasJsonCredentialConfigured() {
+    return System.getenv().containsKey(SERVICE_ACCOUNT_JSON_ENV_VARIABLE);
+  }
+
+  public static CredentialOptions metadataServiceAccountCredential() {
+    return new CredentialOptions(CredentialType.MetadataServiceAccount);
+  }
+
+  public static CredentialOptions noCredential() {
+    LOG.info("Enabling the use of null credentials. This should not be used in production.");
+    return new CredentialOptions(CredentialType.None);
+  }
+
+  public static CredentialOptions p12Credential(String serviceAccount, String keyFile) {
+    return new CredentialOptions(serviceAccount, keyFile);
+  }
+
+  private final CredentialType credentialType;
+  private final String serviceAccount;
+  private final String keyFile;
+
+  private CredentialOptions(CredentialType credentialType) {
+    this.credentialType = credentialType;
+    this.serviceAccount = null;
+    this.keyFile = null;
+  }
+
+  private CredentialOptions(String serviceAccount, String keyFile) {
```

Comment:
Can we have following:
private CredentialOptions(CredentialType credentialType) {
    CredentialOptions(credentialType, null, null);
}

private CredentialOptions(credentialType, serviceAccount, keyFile) {
   ...
}